validations/src/validations.rs

```diff
@@ -215,55 +220,74 @@ fn update_liars(liars: &mut Vec<bool>, item: RadonTypes, condition: bool) -> Opt
     }
 }
 
+/// An `Either`-like structure that covers the two possible return types of the
+/// `evaluate_tally_precondition_clause` method.
+#[derive(Debug)]
+pub enum TallyPreconditionClauseResult {
+    MajorityOfValues(Vec<RadonTypes>, Vec<bool>),
```

Comment:
Docs: what is this `Vec<bool>`? Or add names to fields:
```suggestion
    MajorityOfValues { reveals: Vec<RadonTypes>, liars: Vec<bool> },
```